crates/biome_analyze/src/lib.rs

```diff
@@ -550,14 +581,24 @@ where
                 }
             };
 
-            let (rule, instance) = match kind {
-                SuppressionKind::Everything => (None, None),
-                SuppressionKind::Rule(rule) => (Some(rule), None),
-                SuppressionKind::RuleInstance(rule, instance) => (Some(rule), Some(instance)),
-                SuppressionKind::TopLevel(rule) => (Some(rule), None),
+            let (rule, instance) = match suppression.kind {
+                AnalyzerSuppressionKind::Everything => (None, None),
+                AnalyzerSuppressionKind::Rule(rule) => (Some(rule), None),
+                AnalyzerSuppressionKind::RuleInstance(rule, instance) => {
+                    (Some(rule), Some(instance))
+                }
+                AnalyzerSuppressionKind::TopLevel(rule) => (Some(rule), None),
             };
 
-            is_top_level_suppression |= kind.is_top_level();
+            // We use `text_range` because we want to check the extended range with trivia, comments in particular
```

Comment:
Do I understand correctly it is still valid to have a license comment before the suppression, for instance?

Reply:
Yes it's still valid :) I can add more tests to cover these cases